src/main/java/profile_manager/frameworks_and_drivers/ProfileManagerDataAccess.java

```diff
@@ -0,0 +1,184 @@
+                        profile.getDob(),
+                        profile.getDescription(),
+                        profile.getSocialLinks(),
+                        profile.getSensitiveWords(),
+                        profile.getInterests(),
+                        profile.getGroups(),
+                        profile.getFriends(),
+                        profile.getBlockedUsers(),
+                        profile.getCreationTime());
+                writer.write(line);
+                writer.newLine();
+            }
+
+            writer.close();
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void addGroupToProfile(String userName, String groupId) {
+        List<String> groups = new ArrayList<>(accounts.get(userName).getGroups());
+        groups.add(userName);
+        accounts.get(userName).setGroups(groups);
+        this.save();
+    }
+
+    public void removeGroupFromProfile(String userName, String groupId) {
+        List<String> groups = new ArrayList<>(accounts.get(userName).getGroups());
+        groups.remove(userName);
+        accounts.get(userName).setGroups(groups);
+        this.save();
+    }
+
+    public void addInterestToProfile(String userName, String interest) {
+        List<String> interests = new ArrayList<>(accounts.get(userName).getInterests());
+        interests.add(userName);
+        accounts.get(userName).setInterests(interests);
+        this.save();
+    }
+
+    public void removeInterestFromProfile(String userName, String interest) {
+        List<String> interests = new ArrayList<>(accounts.get(userName).getInterests());
+        interests.remove(userName);
+        accounts.get(userName).setInterests(interests);
+        this.save();
+    }
+
+    public void addFriendToProfile(String userName, String friend) {
+        List<String> friends = new ArrayList<>(accounts.get(userName).getFriends());
+        friends.add(userName);
+        accounts.get(userName).setFriends(friends);
+        this.save();
+    }
+
+    public void removeFriendFromProfile(String userName, String friend) {
+        List<String> friends = new ArrayList<>(accounts.get(userName).getFriends());
+        friends.remove(userName);
+        accounts.get(userName).setFriends(friends);
+        this.save();
+    }
+
+    public void addBlockedUserToProfile(String userName, String blockedUser) {
+        List<String> blockedUsers = new ArrayList<>(accounts.get(userName).getBlockedUsers());
+        blockedUsers.add(userName);
+        accounts.get(userName).setBlockedUsers(blockedUsers);
+        this.save();
+    }
+
+    public void removeBlockedUserFromProfile(String userName, String blockedUser) {
+        List<String> blockedUsers = new ArrayList<>(accounts.get(userName).getBlockedUsers());
+        blockedUsers.remove(userName);
+        accounts.get(userName).setBlockedUsers(blockedUsers);
+        this.save();
+    }
```

Comment:
getblockedUsertoProfile and getSensitiveWordsListtoProfile mean I can get this user's blocked user list and sensitive word list in Profile.csv.